server/services/actions/postmortem_action.py

```
@@ -0,0 +1,272 @@
+        incident_id: The incident to generate a postmortem for
+        custom_instructions: Optional user-customized instructions (layered on top of defaults)
+
+    Returns:
+        run_id of the dispatched action
+
+    Raises:
+        ValueError: If rate limited or generation already in progress
+    """
+    # Check for active generation (row with NULL content = in progress)
+    with db_pool.get_connection() as conn:
+        with conn.cursor() as cur:
```

Ruff (0.15.12)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)